### ISSUE TITLE
Allow b-embed to dynamically load files

### ISSUE THREAD

**Issue by ChrisHarveyHub** [bootstrap-vue/bootstrap-vue#4762]
### Is your feature request related to a problem? Please describe...

I've set up b-embed to monitor a folder of proxy videos and play them out when the corresponding table row has been clicked.  The problem is, my webpage re-aliases the proxy folder and reloads the web page every time a new video is added making it almost impossible to use while updating.

### Describe the solution you'd like

I'd like the ability for b-embed to alias these source paths at runtime instead of compile time so that I can just reference the files on the server only when they're accessed on the webpage without having them compiled with my project.

### Describe alternatives you've considered

I've tried separating the b-embed into it's own component template, but even if that instance of the component is accessing a different file, the webpage still reloads when a new file is added to the proxy directory.

As you can see below, the assetPath is calculated based on the embedSource prop that is passed when the corresponding table row is clicked on the parent component:
ON PARENT COMPONENT:
*when table row is clicked:
```javascript
this.embedSource = require('../../proxies/' + jobItems[0].cname + '/' + jobItems[0].ghouse + '.mp4');
```

ON CHILD EMBED COMPONENT:
```javascript
<template>
    <div id='embededVideo' class='col-lg-12 col-md-12 col-sm-12'>
        <b-embed type="video"  controls>
            <source :src="assetPath" />
        </b-embed>
    </div>
</template>
<style>
    @import '../assets/style.css';
</style>
<script>
    export default {
        name: 'EmbedVideo',
        props: ['embedSource'],
        computed: {
            assetPath () {
                return this.embedSource;  
            }
        }
    }
</script>
```

**Comment by tmorehouse** [comment]
This sounds very use-case specific.

It sounds like you should be generating the URLs to use (either project/server relative or absolute http: or https:) in a json file (or other data structure), and then perform a lookup on that data object to get the URL or path to use.

Perhaps you should use absolute http/https URLs to reference the files on your server, rather than having webpack load them.  Also note, not all users are using webpack.  Some people use the UMD browser build of BootstrapVue, which does not necessarily have have the `require` method.

**Comment by tmorehouse** [comment]
And perhaps instead of requiring the video file, and passing the full file (which is what `require` does), is to use [`require.resolve`](https://www.bennadel.com/blog/3243-using-require-resolve-to-calculate-module-relative-file-paths-in-node-js.htm) to return the resolved _path_ of the file instead of the actual file data, and pass that to the `:src` from of your `<source>` element.

**Comment by tmorehouse** [comment]
Also note that the `<source>` element is a native browser element that `<b-embed>` has no control over.

**Comment by ChrisHarveyHub** [comment]
Ahhh ok in all the documentation I've read it seemed like webpack was an embedded part of vue and bootstrap vue.

All I'm doing is creating a system to use a browser to view proxy files that are stored on a server elsewhere in the company so I've been confused why I can't find more documentation.  This is a pretty common operation in other applications I've created and can be achieved by simply referencing the file paths on the application server but this is the first time I've tried to do it using Vue.

The files are viewable when I use "require" which is great but the issue is that the project recompiles and the broswer reloads any time a new file is added to the folder (which happens multiple times every hour every day as people create more media) because it's trying to basically download the full file rather than just add another reference.  So I'm pretty sure I need to do what you've recommended (use require.resolve()) but when I use that in place of the "require" command, the video files won't load anymore.

And as for the <source> element thank you for that recommendation! Along with all the other misleading documentation I've seen, I was pretty sure that was part of the <b-embed> element but it seems to be working without so thank you again!

**Comment by ChrisHarveyHub** [comment]
Also just FYI, proxy videos are low resolution versions of their originals so really one way to think about it is:
I'm essentially trying to reference a folder of tiny .png files that I'm using as logos for my site but we update the logo daily.  So I want to create a system where I could add logos to a referenced folder whenever I'd like without having to recompile the project to view them.

I thought it would be easy with b-embed but so far it's proven impossible to even view the files without using Require().  There might be something I'm missing about require.resolve() and how I'm using it but right now it seems like even using that, the video still won't load.

**Comment by tmorehouse** [comment]
What is happening in dev mode is webpack hot module reload (HMR) and directory watching.

**Comment by tmorehouse** [comment]
Related: #4743

**Comment by tmorehouse** [comment]
you may want to ask the question at the webpak repo or stackexchange, as this is a webpack related issue, not a BootstrapVue. issue.

try using a plain `<video>` element with nested sources.  you will encounter the same issues:

```html
<video controls>
  <source :src="assetPath" />
</video>
```

**Comment by ChrisHarveyHub** [comment]
The explanation of what's happening in dev mode is exactly what I'm trying to fix.  You mentioned earlier that there should be a way to do this without webpack?

I've removed the nested <source> per your recommendation but it didn't change performance at all.

Thanks for that link btw but what was it for?

I appreciate all this help!  As scottadamsmith mentioned in the link, along with it's obvious functional perks, all this communication is a big part of why I've been loving this library so far.

**Comment by tmorehouse** [comment]
Oops that related link was supposed to be #4741 (your other issue). I've corrected teh coment above.

**Comment by tmorehouse** [comment]
What you may want to do is have your video files in a folder that is rooted under your webserver doc root, and just use URLs that are relative to that, and forgo `require` altogether (so the large files are not imported into the project)

**Comment by ChrisHarveyHub** [comment]
Ah haha that makes sense.

You mean like this?

components (folder)
     proxies (folder)
          *videos here*
     embedPanel.vue

and then reference the path relatively like './proxies/filename.mp4'?

They don't seem to be loading but I'm not getting errors either.

**Comment by ChrisHarveyHub** [comment]
When I do that and inspect the url for the video it's looking for the file at http://localhost:8081/proxies/filename.mp4

**Comment by tmorehouse** [comment]
You would need to configure your router/webserver to handle serving the assets.

usually people create a `/public` or `/assets` directory for the file to be served.

**Comment by tmorehouse** [comment]
Some good resources:

- https://vue-loader.vuejs.org/guide/asset-url.html
- https://stackoverflow.com/questions/47313165/how-to-reference-static-assets-within-vue-javascript
- https://cli.vuejs.org/guide/html-and-static-assets.html

These assume the assets are "static" meaning new assets are not being added dynamically at runtime.

For express and a public folder taht may have content changing at runtime:

- https://alligator.io/nodejs/serving-static-files-in-express/

**Comment by ChrisHarveyHub** [comment]
Ok awesome, thank you! New assets are being added dynamically at runtime and not at any predictable time so I'm hoping that final express solution will be the key to what I'm looking for.

Thanks again :)

**Comment by ChrisHarveyHub** [comment]
I FIGURED IT OUT THANK YOU @tmorehouse!!!!

I noticed the alligator.io article was trying to get me to create the REST service I realized I already had as the backend for my .vue project so I was able to put the files in the exposed public directory from my REST project and then reference that from my .vue project.

Took a while but I fully understand the how and the why of what's going on and it feels real good.  Thanks again!

I can make a little write-up for anyone who would find this to be a useful little article.
